File: 1. Practica/1. Introduccion-POO/ejercicio-5/Persona.py
class Persona:
    def __init__(self, nombre, paterno, materno, edad, ci):
        self.nombre = nombre
        self.paterno = paterno
        self.materno = materno
        self.edad = edad
        self.ci = ci
    def mostrarDatos(self):
        print(f"Nombre: {self.nombre}")
        print(f"Paterno: {self.paterno}")
        print(f"Materno: {self.materno}")
        print(f"Edad: {self.edad}")
        print(f"CI: {self.ci}")

    def mayorDeEdad(self):
        return self.edad >= 18
    def modificarEdad(self, nueva_edad):
        self.edad = nueva_edad
    def mismoPaterno(self, otra_persona):
        return self.paterno == otra_persona.paterno

p1 = Persona("Pedro", "Perez", "Gomez", 20, "8335522")
p2 = Persona("Ana", "Perez", "Huanca", 24, "12345000")
p1.mostrarDatos()
print("Es mayor de edad:", p1.mayorDeEdad())
p2.mostrarDatos()
print("Es mayor de edad:", p2.mayorDeEdad())
print("Tienen el mismo paterno:", p1.mismoPaterno(p2))

File: 1. Practica/1. Introduccion-POO/ejercicio-5/Persona.java
package Introduccion_POO;
public class Persona {
    public  String nombre;
    public String paterno;
    public String materno;
    private int edad;
    private String ci;
    public Persona(String nombre, String paterno, String materno, int edad, String ci) {
        this.nombre = nombre;
        this.paterno = paterno;
        this.materno = materno;
        this.edad = edad;
        this.ci = ci;
    }
    public void mostrarDatos() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Paterno: " + paterno);
        System.out.println("Materno: " + materno);
        System.out.println("Edad: " + edad);
        System.out.println("CI: " + ci);
    }
    public boolean mayorEdad() {
        return edad >= 18;
    }
    public void modificarEdad(int nuevaEdad) {
        if (nuevaEdad >= 0) {
            edad = nuevaEdad;
            System.out.println("Edad modificada: " + edad);
        } else {
            System.out.println("error...");
        }
    }
    public boolean mismoPaterno(Persona otraPersona) {
        return this.paterno.equals(otraPersona.paterno);
    }
   
}
